### PR TITLE
Added SHRDLU and MicroPlanner.

### PR DESCRIPTION
**Pull request by eswenson1**
Note: SHRDLU only works with the visual block rendering on pdp10-ka with the Type 340 display enabled.  Even then, there is some issue that causes the SLAVE interface to the Type 340 display to fail, with an error indicating that the 340 is not available.  This is still to be tracked down.

Without the Type 340 display, SHRDLU should work reasonably well.  It fails at some things that the DEMO apparently succeeded in doing.  It is not clear why, but likely the demo was created at one point, and the SHRDLU sources were updated after that point, breaking some things.

SHRDLU can be invoked with :SHRDLU;SHRDLU.

Once you invoke SHRDLU, you'll see output similar to this:

SHRDLU VERSION 1000   LOADED 2024-07-21 IN BLISP 2156

REFER
COMMENTS AND QUESTIONS TO DDM

VERSION
RECONSTRUCTED BY EJS

-IF YOU ARE NEAR A DEC-340
     DO YOU WANT THE DISPLAY (TYPE " Y  " OR " N  " )

If you are running the pdp10-ka emulator, have enabled the DPY device, and want to see the display, answer "Y". Otherwise, answer "N".

Then, after some more output you will see:

YOU ARE NOW IN A READ-EVAL-PRINT LOOP

TYPE " GO  " TO ENTER READY STATE

>>>

You are now in command mode and can type various commands, such as "SHOW". You can also type "GO" to enter ready mode.  It is in ready mode that you can issue commands like "pick up a big red block.", to which SHRDLU should respond "OK.".  Note that commands should be ended with a period ("."). Questions should be ended with an exclamation mark ("?"). An example question might be:  "what is in the box?".

MicroPlanner can be invoked with :SHRDLU;PLNR.  As a simple test of MicroPlanner, you can enter these expressions (marked with "==>" -- don't type that).  If the (THVAL ...) form results in printing "(FALLIBLE TURING)" then you know that basic goal reaching works.

>>>  TOPLEVEL
LISTENING  THVAL
==> (THASSERT (HUMAN TURING))
((HUMAN TURING))
==> (DEFPROP THEO1 (THCONSE (X) (FALLIBLE $?X) (THGOAL (HUMAN $?X))) THEOREM) THEO1
==> (THASSERT THEO1)
THEO1
==> (THVAL (THGOAL (FALLIBLE TURING) (THTBF THTRUE)) NIL) (FALLIBLE TURING)

### CODE REVIEW

**Comment by eswenson1** [comment]
Note: In order to use the type 340 display, you need an updated LISP;SLAVE FASL.  See this PR for the fix:  https://github.com/PDP-10/its/pull/2319

**Comment by larsbrinkhoff** [comment]
Didn't you make some (bug) fixes to the original files?  For posterity, it would be good to have those edits as commits in the git history.

**Comment by eswenson1** [comment]
Yes, lots of fixes. So you want me to commit the original versions of the files in one commit and then rename/update them to my versions in a second commit?

I may have to abandon this PR and create a new one for that because I have no clue how to do that after the fact.

**Comment by larsbrinkhoff** [comment]
Yes, that's what I want.  I would rather we don't open another pull request but reuse this one.  If you send all the original files, I can do the git time travel.

**Comment by larsbrinkhoff** [comment]
I rebased the branch and put the original files in a commit before @eswenson1's fixed version.  Now it's much easier to see the changes that were made to run SRHDLU on the latest Maclisp.

I verified this on KA10; `PICK UP RED PYRAMID.` works and displays correctly on the 340.

File: src/shrdlu/loader.19 [hunk 1]
```diff
@@ -49,81 +70,100 @@
...
-       (fasload grindef fasl com com)
+       (load '((lisp) grinde fasl))
        'CONSTRUCTION/ COMPLETED) 
 
 (defun loadparser nil
 	(mapc 'loader '(syscom morpho show))
 	(mapc 'loader '(progmr ginter gramar dictio))
 	(loader 'setup)
-	(load parser > dsk shrdl1)		;load is defined on ddm;*load >
+	(load parser > dsk xshrdl)		;load is defined on ddm;*load >
```
Comment (larsbrinkhoff) [inline src/shrdlu/loader.19:119]: Curious what this xshrdl is?

Comment (eswenson1) [reply src/shrdlu/loader.19:119]: It should be SHRDLU. At one point, I had multiple SHRDLU directories (for testing things) and I was loading from XSHRDL. I thought I had changed them all back (was using TECO to edit from an iPhone TELNET connection where GUI was all messed up; and I must have missed some.

**Comment by eswenson1** [comment]
@larsbrinkhoff are you going to approve this PR soon? I’d like to get this merged because I have a bunch more fixes I’d like to commit (in a separate PR).

**Review by larsbrinkhoff — APPROVED** [review]

**Comment by larsbrinkhoff** [comment]
I fixed xshrdl and approved.

**Pull request merged**